src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -211,24 +214,44 @@ public Task<IList<Message>> ReceiveAsync(int maxMessageCount)
         {
             MessagingEventSource.Log.MessageReceiveStart(this.ClientId, maxMessageCount);
 
-            IList<Message> messages = null;
+            IList<Message> processedMessageList = null;
             try
             {
+                IList<Message> unprocessedMessageList = null;
+
                 await this.RetryPolicy.RunOperation(
                     async () =>
                     {
-                        messages = await this.OnReceiveAsync(maxMessageCount, serverWaitTime).ConfigureAwait(false);
+                        unprocessedMessageList = await this.OnReceiveAsync(maxMessageCount, serverWaitTime).ConfigureAwait(false);
                     }, serverWaitTime)
                     .ConfigureAwait(false);
+
+                if (this.AfterEachMessageReceiveTasks == null)
```

Comment:
@SeanFeldman - I think a no-op makes sense here, still working on the best way to do it though.